lib/explorer/shared.ex

```diff
@@ -9,7 +9,16 @@ defmodule Explorer.Shared do
   within lists inside.
   """
   def dtypes do
-    non_list_dtypes = [
+    non_list_dtypes = non_list_types()
+    list_dtypes = for dtype <- non_list_dtypes, do: {:list, dtype}
+    non_list_dtypes ++ list_dtypes
```

Comment:
List types are recursive so this is theoretically incomplete. Instead of doing this list, what if we just append `{:list, :any}` instead?

Reply:
Sure, fine by me 👍

Reply:
So it looks like this is used for tests? I'm going to leave it how it was (I just wrapped the non-list ones into a separate function) and then I think we should revisit it in a future PR.